AsyncFixer/AsyncVoid/AsyncVoidFixer.cs

```diff
@@ -36,17 +36,26 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             var diagnostic = context.Diagnostics.First();
             var diagnosticSpan = diagnostic.Location.SourceSpan;
 
-            var methodDeclaration =
-                root.FindToken(diagnosticSpan.Start).Parent.FirstAncestorOrSelf<MethodDeclarationSyntax>();
+            var node = root.FindNode(diagnosticSpan, getInnermostNodeForTie: true);
+            if (node is LambdaExpressionSyntax)
+            {
+                // For async-void lambda, we do not suggest code fixes.
+                return;
+            }

```

Comment:
I didn't investigate further but in certain cases the code fix threw an exception for me, and I think those were such lamdbdas too, so I guess this would also fix that.

Reply:
I now added a nullable check for the methodDeclaration node to be on the safe side.

Reply:
Great!